lib/carrierwave/uploader/versions.rb

```diff
@@ -133,7 +133,7 @@ def version(name, options = {}, &block)
 
           class_eval <<-RUBY, __FILE__, __LINE__ + 1
             def #{name}
-              versions[:#{name}]
+              version_exists?(:#{name}) ? versions[:#{name}] : versions[:#{name}].class.new(model, mounted_as)
```

Comment:
Lately resolving the condition here. Which would make it more consistent with the current state (active or not) of the version (which fixes https://github.com/carrierwaveuploader/carrierwave/issues/2148 too).

I think it is better to return nil for non-active versions, but returning a blank uploader for backwards compatibility.